ext/libdatadog_api/datadog_ruby_common.c

```diff
@@ -29,7 +29,7 @@ VALUE datadog_gem_version(void) {
 }
 
 __attribute__((warn_unused_result))
-ddog_prof_Endpoint endpoint_from(VALUE exporter_configuration) {
+ddog_Endpoint* endpoint_from(VALUE exporter_configuration) {
   ENFORCE_TYPE(exporter_configuration, T_ARRAY);
```

Comment:
I recently introduced this `datadog_ruby_common.h|.c` because it turns out to be surprisingly hard to get extconf.rb/mkmf to allow us to use headers and c files from a different folder.

(Specifically, to not treat them as an outside library, and e.g. include them in the makefile so that they trigger rebuilds and whatnot).

So I created this .h/.c to contain things that both `libdatadog_api` and `datadog_profiling_native_extension` need. And there's a copy-pasted copy on each folder.

I guess since we need to diverge for this function, my suggestion is to remove it entirely from the common .h/.c: move the libdatadog_api copy to crashtracker.c and the profiler copy to ... libdatadog_helpers.c? http_transport.c?

That way we can keep the common file in-sync with copy-paste :smile:

Reply:
Done!